pandas/core/arrays/categorical.py

```diff
@@ -2570,8 +2566,7 @@ def _get_codes_for_values(values, categories):
     if dtype_equal:
         # To prevent erroneous dtype coercion in _get_data_algo, retrieve
         # the underlying numpy array. gh-22702
-        values = getattr(values, "_ndarray_values", values)
-        categories = getattr(categories, "_ndarray_values", categories)
+        pass
```

Comment:
This is nice on a first pass at the diff, but probably better just to blow this branch away